### PR TITLE
feature/signedupload: Support s3

### PR DESCRIPTION
**Pull request by zdyj3170101136**

### CODE REVIEW

**Comment by zdyj3170101136** [comment]
the golangci-lint is not success because of s3 code.
i copy it directly from objstore.
should i changed it?
@brancz 

截屏2023-03-24 上午11 21 49

**Comment by v-thakkar** [comment]
Drive-by comment: Is S3 package written by you or you're using Thanos's package with some modifications? Either way, you'll need to add Parca License in all.

**Comment by zdyj3170101136** [comment]
> Details

you can see the first commit.
the s3 package is init from objstore.
i only add SignedPUT function on s3.Bucket struct.

seems the three auto commit add parca license.

**Comment by zdyj3170101136** [comment]
> Drive-by comment: Is S3 package written by you or you're using Thanos's package with some modifications? Either way, you'll need to add Parca License in all.

you can see the first commit.
the s3 package is init from objstore.
i only add SignedPUT function on s3.Bucket struct.

seems the three auto commit add parca license.

File: cmd/parca/main.go [hunk 1]
```diff
@@ -36,6 +37,7 @@ func main() {
 	ctx := context.Background()
 	flags := &parca.Flags{}
 
+	debug.PrintStack()
```
Comment (kakkoyun) [inline cmd/parca/main.go:40]: Shall we hide this behind a debug flag?

**Review by kakkoyun — COMMENTED** [review]
Thanks for taking care of it 🙏 🤗

I would rather see the changes committed to https://github.com/thanos-io/objstore
We can be better open-source citizens. I could help you with the reviews. I'm still a maintainer of that repo.

**Comment by brancz** [comment]
I’m not convinced this can land in the objstore package. I thought about this when introducing the signed uploads originally, but the reality is that these features work wildly different on each provider so I don’t think an abstraction that’s suitable for the objstore package is possible.

**Comment by zdyj3170101136** [comment]
> Thanks for taking care of it 🙏 🤗
> 
> I would rather see the changes committed to https://github.com/thanos-io/objstore We can be better open-source citizens. I could help you with the reviews. I'm still a maintainer of that repo.

as @brancz said, it is not easy to merge this to objstore

**Pull request closed without merging**